pkg/querier/querier.go

```diff
@@ -359,6 +359,7 @@ func (q querier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Mat
 	sets := make(chan storage.SeriesSet, len(q.queriers))
 	for _, querier := range q.queriers {
 		go func(querier storage.Querier) {
+			// We should always select sorted here as we will need to merge the series
```

Comment:
This comment is a bit misleading because I think what you meant that is "if we need to query multiple backend, then we need to merge results which depends on the result from each querier being sorted. We are not sure if each result is sorted default, so it's safer to force sort to ensure the merge can work properly"